### PR TITLE
Have a backup plan when lo_import fails

### PR DESCRIPTION
**Pull request by jlnr**
This gem is a lifesaver! 👍 I'm following the [Rails guide for Active Storage](http://guides.rubyonrails.org/active_storage_overview.html#transforming-images) and uploading and displaying unmodified images works fine.

However, a soon as I try to use variants with MiniMagick, `lo_import` fails with the unhelpful error message "PG::Error (ERROR:  invalid large-object descriptor: 0)", both locally, and on Heroku.

I've used byebug to verify that the path that's being passed to `lo_import` is correct, and I can `lo_import(...)` the temporary file from a psql console. I suspect that it has something to do with permissions, as `Dir.tmpdir` on macOS returns a directory that only the current user has access to, and which the postgres server process can't touch. However I couldn't work around it even when I monkeypatched `Dir.tmpdir` to return `/tmp`.

Long story short, the gem already has fallback logic when `io.respond_to?(:to_path)` doesn't apply. This PR rearranges the code so that this code path is also tried if the `lo_import` fast path doesn't work, which fixes the issue for me.

### CODE REVIEW

**Comment by lsylvester** [comment]
Hi @jlnr, 

Thanks for your PR. I am glad you are finding this project useful.

I have investigated this issue and believe I have found the cause. All of the lo* operations need to happen inside of a transaction, and it they are used without a transaction this is the error that gets raised.

The upload code was missing the transaction and I missed it in the tests because all of the tests are automatically wrapped in a transaction.

I have made a fix in master. Are you able to test the updated code and confirm if it resolves your issue?

Thanks.

**Comment by jlnr** [comment]
Thanks! Works both locally and on Heroku.
